### DISCUSSION TITLE
ValueError: Length mismatch: Expected 1050 rows, received array of length 2170 when trying to cluster cells

### DISCUSSION THREAD

**Discussion by whitneyt1** [angelolab/ark-analysis#709] [Q&A] [answered]
When clustering my cells after segmentation then pixel clustering, I receive this error. ValueError: Length mismatch: Expected 1050 rows, received array of length 2170 

<img width="1547" alt="cellclusteringerror" src="https://user-images.githubusercontent.com/46573884/190476902-37df7183-a1ff-4e35-998a-2f8c55ea30d3.PNG">

The only things I adjusted were my paths to specific directories, and the path to the cell_table_size_normalized.csv. 

<img width="1432" alt="cellclusteringparams" src="https://user-images.githubusercontent.com/46573884/190476908-9ec3ceec-d820-4318-a3b5-b9e594b25823.PNG">

My cell_table dimensions are 
`<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20122 entries, 0 to 20121
Data columns (total 64 columns):` 
<img width="321" alt="dfdims" src="https://user-images.githubusercontent.com/46573884/190478922-b9a8fa48-6784-4870-8bc4-72ba0b408d73.PNG">

Any advice if I am applying the wrong cell_table or other table with incorrect dimensions? Thank you for all your help!

**Comment by whitneyt1** [comment]
@alex-l-kong any advice?

**Reply by alex-l-kong** [reply]
@whitneyt1 highly unlikely but I first want to verify (and hopefully remove) the possibility that you have duplicate segmentation labels for at least one FOV in your cell table. Can you verify this by running in the notebook:

```
import pandas as pd
cell_table = pd.read_csv(cell_table_path)
for fov in fovs:
    cell_table_sub = cell_table[cell_table['fov'] == fov].copy()
    print(cell_table_sub['label'].value_counts().head(5))  # if this fails change 'label' to 'segmentation_label'
```

A valid cell table should not have any segmentation label with a count of 2 or more.

**Reply by whitneyt1** [reply]
All of my values are 1! 
<img width="120" alt="checkcellabels" src="https://user-images.githubusercontent.com/46573884/191083395-47d34841-cbb3-4268-a857-adfb609fd919.PNG">

**Reply by alex-l-kong** [reply]
@whitneyt1 hmm, seems like you have a lot of duplicate cell labels of 0 in each FOV too, which is likely causing the error. The easiest solution is to directly remove these label 0 cells from your cell table, they're likely background noise.

Can you also ensure you have the most up-to-date version of our code (`git pull`)? Could be related to a bug we had in the past.

**Reply by whitneyt1** [reply]
great. I updated and am still receiving the errors. 

Can I use `cell_table= cell_table[cell_table['label'] != 0]` to remove 0s? Do you have advice on preventing as many 0 labels in the future? thank you!

**Reply by ngreenwald** [reply]
I would say wait until we've released the updated version of the codebase this week that resolves the size-zero cells error. That way we can be sure we're not troubleshooting something that's already been fixed

**Reply by HPiyadasa** [reply]
[@ngreenwald] I am also getting the similar issue with cell labels. Of the 32016 rows from a test set, I am seeing 16338 are labelled as "0". And then the remaining cell labels also skip numbers and there are duplicate labels. See attached. 


<img width="1470" alt="Screen Shot 2022-09-21 at 2 54 43 PM" src="https://user-images.githubusercontent.com/62519486/191598475-0335880d-547e-4347-9816-7fd636fcd3d8.png">

**Comment by ngreenwald** [comment, marked as answer]
We've just made a new release that addresses this issue. Once you [follow the instructions](https://github.com/angelolab/ark-analysis#updating-the-repository) for updating, this should be resolved. I would recommend starting from the beginning of the whole pipeline in a clean folder to make sure you're using only updated files for all steps.
